### DISCUSSION TITLE
Guidelines for distribution packaging?

### DISCUSSION THREAD

**Discussion by MingcongBai** [rustdesk/rustdesk#9796] [General]
Hi all!

I'm a maintainer of [AOSC OS](https://aosc.io) and am interested in packaging RustDesk for our distro. Upon checking the licensing documents, I seem to have found inconsistent/conflicting information about redistribution. From this repository, we see that the source is licensed under AGPL-3.0, however, from the home page, we see a different set of [Terms and Conditions](https://rustdesk.com/terms/), which specifies:

> Company grants you a limited, non-exclusive, non-transferable, revocable license to use the Software for your personal or internal business use, subject to the terms and conditions of this Agreement. You are not allowed to rent, lease, lend, sell, redistribute, or sublicense the Software or any part of it.

Can anyone please clarify on whether this applies to RustDesk copies built from source, or that it only applies to the pre-built binaries available from the home page?

Thanks in advance.

**Comment by xlionjuan** [comment]
~~Is it support `.deb`?~~ Yes
https://aosc.io/oma

**Reply by MingcongBai** [reply]
Yes, we are a .deb based distribution but we are not a Debian-derivative.

As for oma, we are working to make it work on more .deb-based distros.

**Reply by KexyBiscuit** [reply]
To be clear, yes, AOSC OS provides compatibility with .deb designed for Debian and its derivatives, via [Spiral](https://www.bilibili.com/read/cv33454741/).

However, like what @MingcongBai mentions, AOSC OS is not a Debian derivative, so we prefer to package all open source or source available applications by ourselves, prevent the usage of Spiral as much as possible.

So, we prefer not to ask the user to get .deb from the official release page, but to provide our own source-built version instead.

**Reply by xlionjuan** [reply]
> To be clear, yes, AOSC OS provides compatibility with .deb designed for Debian and its derivatives, via [Spiral](https://www.bilibili.com/read/cv33454741/).

Got it.

> So, we prefer not to ask the user to get .deb from the official release page, but to provide our own source-built version instead.

Since the complexity of the Flutter, there is no any places distribute RustDesk with "build from source", except Arch AUR's PKGBUILD, even Flathub is built from `.deb`.

<hr>

> we see a different set of [Terms and Conditions](https://rustdesk.com/terms/), which specifies:
>> Company grants you a limited, non-exclusive, non-transferable, revocable license to use the Software for your personal or internal business use, subject to the terms and conditions of this Agreement. You are not allowed to rent, lease, lend, sell, redistribute, or sublicense the Software or any part of it.

I guess it is for Pro server, the ToS should have better writing/elaborate.
